### PR TITLE
fix: preselect ticket buyer as holder for existing buyers

### LINKED ISSUE #5851: Preselect "Ticket holder" same as "Ticket Buyer
Always preselect "Ticket holder is the same as ticket buyer." by default, but keep functionality as is. So, the ticket buyer can untick the box if they want.

Screenshot from 2020-12-01 20-43-03

### PR DESCRIPTION
**Pull request by Tlazypanda**
Fixes #5851 

#### Short description of what this resolves:
Preselect ticket buyer as holder for exisiting buyers i.e except new buyers for other cases ticket buyer is preselected as ticket holder

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Tlazypanda** [comment]
@iamareebjamal can you please review? :sweat_smile:

**Comment by iamareebjamal** [comment]
If last name is null?

**Comment by Tlazypanda** [comment]
@iamareebjamal is this okay now?

File: app/components/forms/orders/order-form.js [hunk 1]
```diff
@@ -25,13 +25,16 @@ export default Component.extend(FormMixin, {
...
   holders           : computed('data.attendees', 'buyer', function() {
-    this.data.attendees.forEach(attendee => {
-      attendee.set('firstname', '');
-      attendee.set('lastname', '');
-      attendee.set('email', '');
-      attendee.set('acceptVideoRecording', true);
-      attendee.set('acceptShareDetails', true);
-      attendee.set('acceptReceiveEmails', true);
+    this.data.attendees.forEach((attendee, index) => {
+      if (index === 0 && this.buyerFirstName!==null && this.buyerLastName!=null) {
```
Comment (maze-runnar) [inline app/components/forms/orders/order-form.js:29]: ```if (index === 0 && this.buyerFirstName && this.buyerLastName) ```

**Comment by maze-runnar** [comment]
It's working fine, but I have a doubt regarding here - 
Screenshot from 2021-05-14 22-09-37

ticket holder is same person as ticket buyer, this should be checked by default, or should be removed.

**Comment by MeghalBisht** [comment]
> ticket holder is same person as ticket buyer, this should be checked by default, or should be removed.

Checked by default for the first ticket

**Comment by maze-runnar** [comment]
> > ticket holder is same person as ticket buyer, this should be checked by default, or should be removed.
> 
> Checked by default for the first ticket

ok, please add this.

**Comment by MeghalBisht** [comment]
@maze-runnar Please check it works for me now.

File: app/components/forms/orders/order-form.js [hunk 2]
```diff
@@ -42,7 +45,12 @@ export default Component.extend(FormMixin, {
     }
     return true;
   }),
-  sameAsBuyer: false,
+  sameAsBuyer: computed('data', function() {
+    if (!this.buyerHasFirstName || !this.buyerHasLastName) {
+      return false;
+    }
+    return true;
```
Comment (maze-runnar) [inline app/components/forms/orders/order-form.js:52]:
```suggestion
   return (this.buyerHasFirstName && this.buyerHasLastName);

```

Comment (maze-runnar) [inline app/components/forms/orders/order-form.js:52]:
```suggestion
    return (this.buyerHasFirstName && this.buyerHasLastName);
```

**Review by maze-runnar — APPROVED** [review]

**Review by MeghalBisht — APPROVED** [review]

**Pull request merged**